### DISCUSSION TITLE
Performance issues linked to backgrounds (of all things)

### DISCUSSION THREAD

**Discussion by AntonPetrochenko** [ppy/osu#29880] [Q&A]
To preface this, I'm as confused by this as you may be.

So I've been having these bizarre FPS drops on 2024.906.2, and while troubleshooting them I isolated the cause to be regular old png backgrounds.

[DxDiag.txt](https://github.com/user-attachments/files/17003788/DxDiag.txt) (osu running on the 1650, driver version 561.09)
[compressed-logs.zip](https://github.com/user-attachments/files/17003789/compressed-logs.zip)

Essentially:

- Certain maps have various degrees of FPS drops on my machine
- FPS drops appear to be somewhat tied to the resolution/file size of the background image
- Removing the background from an offending map solves the issue completely
Additionally, setting background dim to 100% doesn't change anything (and bizarrely, causes a constant 50% performance drop on OpenGL renderer), only outright deleting the background image file works.

**Comment by peppy** [comment]
Sounds like hitting fill rate / draw call / gpu limits.

**Reply by AntonPetrochenko** [reply]
How would one go about troubleshooting this? Don't know the first thing about rendering but would love to help

**Reply by peppy** [reply]
A video with frame graphs up would help confirm this (Ctrl+F11 twice in game). You could also check your gpu usage in task manager while playing if you're on a newer windows version.
